pkgs/tools/audio/openai-whisper-cpp/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, SDL2
+, makeWrapper
+, wget
+, Accelerate
+, CoreGraphics
+, CoreVideo
+}:
+
+stdenv.mkDerivation rec {
+  pname = "whisper.cpp";
+  version = "1.0.4";
+
+  src = fetchFromGitHub {
+    owner = "ggerganov";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-lw+POI47bW66NlmMPJKAkqAYhOnyGaFqcS2cX5LRBbk=";
+  };
+
+  # The upstream download script tries to download the models to the
+  # directory of the script, which is not writable due to being
+  # inside the nix store. This patch changes the script to download
+  # the models to the current directory of where it is being run from.
+  patches = [ ./download-models.patch ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  buildInputs = [ SDL2 ] ++ lib.optionals stdenv.isDarwin [ Accelerate CoreGraphics CoreVideo ];
+
+  buildPhase = ''
+    make main
+    make stream
+  '';
```

Comment:
Why not use makeFlags?

Reply:
also, `runHook preBuild` + `runHook postBuild` (thanks for mentioning this Sandro!)